### ISSUE TITLE
Handle overpays/refunds

### ISSUE THREAD

**Issue by loredanacirstea** [raiden-network/microraiden#172]
Micropayment with a higher value than required (-> higher balance) is accepted by the server.
Decide how this will be handled in case it happens.

**Comment by andrevmatos** [comment]
@pcppcp What do you think of an algorithm like this, on the server:
```
# for each channel, store not only last paid BP, but also the "spent" amount
if invalid(BP):
    return 402 (price)
if BP > saved_BP:
    save(BP)
if price > (BP - spent):
    return 402 (price - (BP - spent)) # ask only for missing fraction
spent += price
return 200
```
This way, usually BP and spent will be equal, which will cause a 402 answer with the current price as amount. If the user pays, e.g., 1.5*price, request will go through, and next one will ask for only half the price. If the user pays only 0.5*price, they'll receive 402 again, asking for the other half. If the user sends 10*price, they'll be able to do 10 requests without being presented a 402 (overpay). If the channel is closed in meantime, this overpay can't be refunded, and the receiver will keep the "change".
It should also fix #173

**Comment by pcppcp** [comment]
@andrevmatos sure that's also doable, but it involves another state saving on the side of the proxy. I don't want to do major changes/refactoring before the release, but it's definitely a feature we can discuss.
I'd prefer to not include such functionality in channel manager (aka "core" of microraiden), which provides the very basic functionality and its code should be kept as minimal as possible.

A fix for #173 is coming soon, server will simply not register payment if `BP.price < resource.price` and will send back an error header.

**Comment by pcppcp** [comment]
fixed by PR 177
